### PR TITLE
add more logs for super catchup

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
This PR added state_hashes to "Catchup state" and added a bunch of logs for `Super_catchup.initial_validate` function.

### CODE REVIEW

File: src/lib/transition_frontier/full_catchup_tree.ml [hunk 1]
```diff
@@ -116,7 +116,7 @@ end
    ancestor is at most k *)
 type t =
   { nodes: Node.t State_hash.Table.t
-  ; states: int Node.State.Enum.Table.t
+  ; states: (int * State_hash.Set.t) Node.State.Enum.Table.t
```
Comment (mrmr1993) [inline src/lib/transition_frontier/full_catchup_tree.ml:119]: Nit: it's good practice to add a record to identify the fields if the type isn't self-documenting

File: src/lib/mina_commands/mina_commands.ml [hunk 2]
```diff
@@ -356,7 +356,7 @@ let get_status ~flag t =
     in
     match Transition_frontier.catchup_tree frontier with
     | Full full ->
-        Some (Hashtbl.to_alist full.states)
+        Some (List.map (Hashtbl.to_alist full.states) ~f:(fun (state, (n, _hashes)) -> (state, n)))
```
Comment (mrmr1993) [inline src/lib/mina_commands/mina_commands.ml:359]: This seems to be changing the CLI output. If you plan to merge this, can you include the old vs new output for the affected command, and for completeness can you confirm that the output of the GraphQL endpoint for status is unchanged?

File: src/lib/transition_frontier/full_catchup_tree.ml (hunk 1 above)
Comment (ghost-not-in-the-shell) [reply src/lib/transition_frontier/full_catchup_tree.ml:119]: After thinking about it twice, I think I just need the set of state hashes there. So I would change the type to be:
```
; states: State_hash.Set.t Node.State.Enum.Table.t
```
This is much simpler.

File: src/lib/mina_commands/mina_commands.ml (hunk 2 above)
Comment (ghost-not-in-the-shell) [reply src/lib/mina_commands/mina_commands.ml:359]: This part should convert the `catchup_states` to the original format. I would run a node to test it.

**Pull request closed without merging**